### PR TITLE
DevTools: Add ToolsController, BeginAction, and GitAttributesAction

### PR DESCRIPTION
**Pull request by labkey-adam**
Clean up platform .gitattributes file a bit

### CODE REVIEW

File: devtools/src/org/labkey/devtools/ToolsController.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+
+    @RequiresPermission(AdminPermission.class)
+    public class GitAttributesAction extends SimpleViewAction<Object>
+    {
+        @Override
+        public ModelAndView getView(Object o, BindException errors)
+        {
+            return new VBox(
+                new GitAttributesView("core"),
```
Comment (labkey-jeckels) [inline devtools/src/org/labkey/devtools/ToolsController.java]: Why these two modules but no others?

Comment (labkey-adam) [reply devtools/src/org/labkey/devtools/ToolsController.java]: These modules are simply used to locate the root of the platform and commonAssays repos, respectively. The .gitattributes files live at the repo root, not at the module level. These modules were picked somewhat arbitrarily... modules that are likely to stick around in those repos for the foreseeable future. These are the only repos with .gitattributes files that need to be managed (i.e., that include an explicit list of files). I'll add a comment to that effect with my next platform commit.

Comment (labkey-jeckels) [reply devtools/src/org/labkey/devtools/ToolsController.java]: Got it. A comment sounds great.

**Comment by labkey-adam** [comment]
This action "validates" our two non-trivial .gitattributes files by outputting all filenames listed that don't actually exist, thereby highlighting files that have been moved, renamed, or deleted. For example, it pointed out the test JSP files that Matt moved into devtools, making it easy to update .gitattributes with their new locations.

**Comment by labkey-adam** [comment]
I wrote this as a command line tool to help during the assay refactor and Rosaline used it for her Microarray refactor, but building it into devtools will make it more accessible. I'm still not convinced these files are helping with line endings, but if we're going to have them we might as well have an easy way to maintain them.

**Pull request merged**